pkg/apis/rollouts/v1alpha1/analysis_types.go

```diff
@@ -134,6 +135,15 @@ func (as AnalysisPhase) Completed() bool {
 	return false
 }
 
+// DatadogMetric defines the datadog query to perform canary analysis
+type DatadogMetric struct {
+	// ApiKey and AppKey authenticate with the datadog API (all requests must be authenticated)
+	ApiKey string `json:"apiKey"`
+	AppKey string `json:"appKey"`
```

Comment:
We should follow the wavefront approach for sensitive keys: https://github.com/argoproj/argo-rollouts/blob/master/metricproviders/wavefront/wavefront.go#L156-L191

Reply:
Good call!

In regards to fetching secrets for the Datadog metric provider, I made a couple assumptions:

- I thought the API Keys would be able to be passed into the AnalysisTemplate using args/params, but that's probably not necessary if we require a datadog secret to exist in the argo-rollouts namespace
- The Secret name for datadog API keys will be "datadog-api-keys"?
- The api and app keys will be named "api-key" and "app-key"?

Reply:
Settled on:

- Secret name: `datadog-api-keys`
- api key: `datadog-api-key`
- app key: `datadog-app-key`